storm-server/src/test/java/org/apache/storm/scheduler/resource/strategies/scheduling/TestLargeCluster.java

```diff
@@ -0,0 +1,458 @@
+        return in == null ? ClassLoader.getSystemClassLoader().getResourceAsStream(resource) : in;
+    }
+
+    /**
+     * Read the contents of the fully qualified resource path.
+     *
+     * @param resource
+     * @return
+     * @throws Exception
+     */
+    public static byte[] getResourceAsBytes(String resource) throws Exception {
+        InputStream in = getResourceAsStream(resource);
+        if (in == null) {
+            return null;
+        }
+        try (ByteArrayOutputStream out = new ByteArrayOutputStream()) {
+            while (in.available() > 0) {
+                out.write(in.read());
+            }
+            return out.toByteArray();
+        }
+    }
+
+    public static ClassLoader getContextClassLoader() {
+        return Thread.currentThread().getContextClassLoader();
+    }
+
+    /**
+     * Create an array of TopologyDetails by reading serialized files for topology and configuration in the
+     * resource path.
+     *
+     * @param failOnParseError throw exception if there are unmatched files, otherwise ignore unmatched and read errors.
+     * @return An array of TopologyDetails representing resource files.
+     * @throws Exception
+     */
+    public static TopologyDetails[] createTopoDetailsArray(boolean failOnParseError) throws Exception {
+        List<TopologyDetails> topoDetailsList = new ArrayList<>();
+        List<String> errors = new ArrayList<>();
+        List<String> resources = getResourceFiles(TEST_RESOURCE_PATH);
+        Map<String, String> codeResourceMap = new TreeMap<>();
+        Map<String, String> confResourceMap = new HashMap<>();
+        for (int i = 0 ; i < resources.size() ; i++) {
+            String resource = resources.get(i);
+            int idxOfSlash = resource.lastIndexOf("/");
+            int idxOfDash = resource.lastIndexOf("-");
+            String nm = idxOfDash > idxOfSlash ? resource.substring(idxOfSlash + 1, idxOfDash) : resource.substring(idxOfSlash + 1, resource.length() - 7);
+            if (resource.endsWith("code.ser")) {
+                codeResourceMap.put(nm, resource);
+            } else if (resource.endsWith("conf.ser")) {
+                confResourceMap.put(nm, resource);
+            } else {
+                LOG.info("Ignoring unsupported resource file " + resource);
+            }
+        }
+        String[] examinedConfParams = {
+                Config.TOPOLOGY_NAME,
+                Config.TOPOLOGY_SCHEDULER_STRATEGY,
+                Config.TOPOLOGY_PRIORITY,
+                Config.TOPOLOGY_WORKERS,
+                Config.TOPOLOGY_WORKER_MAX_HEAP_SIZE_MB,
+                Config.TOPOLOGY_SUBMITTER_USER,
+                Config.TOPOLOGY_ACKER_CPU_PCORE_PERCENT,
+                Config.TOPOLOGY_ACKER_RESOURCES_OFFHEAP_MEMORY_MB,
+                Config.TOPOLOGY_ACKER_RESOURCES_ONHEAP_MEMORY_MB,
+        };
+
+        for (String nm : codeResourceMap.keySet()) {
```

Comment:
It looks like `nm` is actually `topoId`? What does `nm` stand for?

Reply:
nm (name) is the crud that stands in from of the code.ser and conf.ser in the file name and is the same for two topology and config files for the same topology. The code is expecting/assuming topology name to be embedded in the file names.

Reply:
Thanks.  I was confused about `nm` variable. Maybe `name` is more clear. But that's fine.